From 4f49d99b8381a580342c0283312e6452d972d374 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 12 Sep 2013 13:26:30 +0800
Subject: [PATCH 11/45] ENGR00276661 GPS:Delete unnecessary switchUser call

If userId doesn't change, no need to do switchuser.or it will
fastly disable and enable gps in it.For some gps modules, enable
immediately after disable won't work well.

Change-Id: I1638745677c0f8dc0ce7e4712e654cba06cdf7f7
Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 services/core/java/com/android/server/LocationManagerService.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/LocationManagerService.java b/services/core/java/com/android/server/LocationManagerService.java
index 468ead0..44e4533 100644
--- a/services/core/java/com/android/server/LocationManagerService.java
+++ b/services/core/java/com/android/server/LocationManagerService.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2007 The Android Open Source Project
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -313,7 +314,10 @@ public class LocationManagerService extends ILocationManager.Stub {
             public void onReceive(Context context, Intent intent) {
                 String action = intent.getAction();
                 if (Intent.ACTION_USER_SWITCHED.equals(action)) {
-                    switchUser(intent.getIntExtra(Intent.EXTRA_USER_HANDLE, 0));
+                    int newUserId = intent.getIntExtra(Intent.EXTRA_USER_HANDLE, 0);
+                    if (mCurrentUserId != newUserId) {
+                        switchUser(newUserId);
+                    }
                 } else if (Intent.ACTION_MANAGED_PROFILE_ADDED.equals(action)
                         || Intent.ACTION_MANAGED_PROFILE_REMOVED.equals(action)) {
                     updateUserProfiles(mCurrentUserId);
-- 
1.8.0

